src/actions/callactivitywithretryaction.ts

```diff
@@ -0,0 +1,15 @@
+import { ActionType, IAction, RetryOptions } from "../classes";
+
+export class CallActivityWithRetryAction implements IAction {
+    public readonly actionType: ActionType = ActionType.CallActivityWithRetry;
+
+    constructor(
+        public readonly functionName: string,
+        public readonly retryOptions: RetryOptions,
+        public readonly input?: unknown,
+    ) {
+        if (!retryOptions) {
```

Comment:
why do you only validate retryOptions?

Reply:
Good point; we could validate the functionName too as it's passed in from the user code.